modules/nlu/src/backend/training-pipeline.ts

```diff
@@ -97,7 +101,16 @@ const PreprocessInput = async (input: TrainInput, tools: Tools): Promise<TrainSt
     makeListEntityModel(list, input.botId, input.languageCode, tools)
   )
 
-  const intents = await ProcessIntents(input.intents, input.languageCode, list_entities, tools)
+  const intents = await ProcessIntents(
+    input.intents,
+    input.languageCode,
+    list_entities,
+    input.list_entities,
+    input.pattern_entities,
```

Comment:
that keeps bugging me, can you add a comment that says why we pass input and output list_entities